rioxarray/rioxarray.py

```diff
@@ -904,21 +904,28 @@ def isel_window(self, window):
         ----------
         window: :class:`rasterio.window.Window`
             The window of the dataset to read.
+        pad: bool
+            Set to True to expand returned DataArray to dimensions of the window
 
         Returns
         -------
         :obj:`xarray.Dataset` | :obj:`xarray.DataArray`:
             The data in the window.
         """
         (row_start, row_stop), (col_start, col_stop) = window.toranges()
-        row_start = math.ceil(row_start) if row_start < 0 else math.floor(row_start)
+        row_start = 0 if row_start < 0 else math.floor(row_start)
         row_stop = math.floor(row_stop) if row_stop < 0 else math.ceil(row_stop)
-        col_start = math.ceil(col_start) if col_start < 0 else math.floor(col_start)
+        col_start = 0 if col_start < 0 else math.floor(col_start)
         col_stop = math.floor(col_stop) if col_stop < 0 else math.ceil(col_stop)
         row_slice = slice(int(row_start), int(row_stop))
         col_slice = slice(int(col_start), int(col_stop))
+        array_subset = self._obj
+        if pad:
+            array_subset = array_subset.rio.pad_box(
```

Comment:
Mind trying the pad afterwards? If it works, it should be a more efficient & performant operation since it is working on a smaller array.

Reply:
it does seem to work now that isel can't return a malformed shape, but it returns a slightly larger subset (1x16x16 vs the expected 1x15x15 for the first test result, and 1x16x15 for the second test result) . Weirdly, it still passes the asserts I have in the tests even though the result of `res1 == exp1` is a shapeless and empty, despite that calling `.all()` on it returns True. This is likely due to some behavior in pad_box/pad_xy. I guess the question is: is this considered a bug within pad_box/pad_xy and/or is it sufficiently harmless to ignore for this pr.

Reply:
It may have been doing this earlier as well, so I'll go ahead and push the changes I have to do the padding afterwards.

Reply:
Sounds harmless to me. I think it is related to #155.

Reply:
okay that sounds right, and can be resolved in a different pr